pkg/virt-launcher/virtwrap/api/schema_test.go

```diff
@@ -406,3 +406,208 @@ var _ = ginkgo.Describe("JSON marshal of the alias of a domain device", func() {
+			Expect(xmlStr).To(ContainSubstring(`<dhCert>test-dh-cert</dhCert>`))
+			Expect(xmlStr).To(ContainSubstring(`<session>test-session</session>`))
+			Expect(xmlStr).To(ContainSubstring(`<cbitpos>51</cbitpos>`))
+			Expect(xmlStr).To(ContainSubstring(`<reducedPhysBits>1</reducedPhysBits>`))
+		})
+
+		ginkgo.It("should unmarshal SEV-SNP launch security with all fields", func() {
+			xmlData := `<LaunchSecurity type="sev-snp" authorKey="yes" vcek="yes"><policy>0x30000</policy><idAuth>test-id-auth</idAuth><idBlock>test-id-block</idBlock><hostData>test-host-data</hostData><dhCert>test-dh-cert</dhCert><session>test-session</session><cbitpos>51</cbitpos><reducedPhysBits>1</reducedPhysBits></LaunchSecurity>`
+
+			var launchSecurity LaunchSecurity
+			err := xml.Unmarshal([]byte(xmlData), &launchSecurity)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(launchSecurity.Type).To(Equal("sev-snp"))
+			Expect(launchSecurity.Policy).To(Equal("0x30000"))
+			Expect(launchSecurity.AuthorKey).To(Equal("yes"))
+			Expect(launchSecurity.VCEK).To(Equal("yes"))
+			Expect(launchSecurity.IdAuth).To(Equal("test-id-auth"))
+			Expect(launchSecurity.IdBlock).To(Equal("test-id-block"))
+			Expect(launchSecurity.HostData).To(Equal("test-host-data"))
+			Expect(launchSecurity.DHCert).To(Equal("test-dh-cert"))
+			Expect(launchSecurity.Session).To(Equal("test-session"))
+			Expect(launchSecurity.Cbitpos).To(Equal("51"))
+			Expect(launchSecurity.ReducedPhysBits).To(Equal("1"))
+		})
+
+		ginkgo.It("should marshal SEV-SNP launch security with minimal fields", func() {
+			launchSecurity := &LaunchSecurity{
+				Type:   "sev-snp",
+				Policy: "0x30000",
+			}
+
+			xmlBytes, err := xml.Marshal(launchSecurity)
+			Expect(err).ToNot(HaveOccurred())
+
+			expectedXML := `<LaunchSecurity type="sev-snp"><policy>0x30000</policy></LaunchSecurity>`
+			Expect(string(xmlBytes)).To(Equal(expectedXML))
+		})
+
+		ginkgo.It("should unmarshal SEV-SNP launch security with minimal fields", func() {
+			xmlData := `<LaunchSecurity type="sev-snp"><policy>0x30000</policy></LaunchSecurity>`
+
+			var launchSecurity LaunchSecurity
+			err := xml.Unmarshal([]byte(xmlData), &launchSecurity)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(launchSecurity.Type).To(Equal("sev-snp"))
+			Expect(launchSecurity.Policy).To(Equal("0x30000"))
+			Expect(launchSecurity.AuthorKey).To(BeEmpty())
+			Expect(launchSecurity.VCEK).To(BeEmpty())
+			Expect(launchSecurity.IdAuth).To(BeEmpty())
+			Expect(launchSecurity.IdBlock).To(BeEmpty())
+			Expect(launchSecurity.HostData).To(BeEmpty())
+		})
+
+		ginkgo.It("should handle empty structure", func() {
+			launchSecurity := &LaunchSecurity{
+				Type: "sev-snp",
+			}
+
+			xmlBytes, err := xml.Marshal(launchSecurity)
+			Expect(err).ToNot(HaveOccurred())
+
+			expectedXML := `<LaunchSecurity type="sev-snp"></LaunchSecurity>`
+			Expect(string(xmlBytes)).To(Equal(expectedXML))
+		})
+	})
+
+	ginkgo.Context("Domain with SEV-SNP launch security", func() {
```

Comment:
Same for these cases, I don't think they add much value. Why not merge them with the above test cases?